packages/magento-product-configurable/components/ConfigurableProductUrls/ConfigurableProductUrls.tsx

```diff
@@ -0,0 +1,40 @@
+import {
+  ProductLinkProps,
+  useProductLink,
+} from '@graphcommerce/magento-product/hooks/useProductLink'
+import { useRouter } from 'next/router'
+import { useEffect } from 'react'
+import { ConfigurableOptionsFragment } from '../../graphql/ConfigurableOptions.gql'
+import { useConfigurableOptionsSelection } from '../../hooks/useConfigurableOptionsSelection'
+
+type ConfigurableProductUrlsProps = {
+  index?: number
+  product: Partial<ConfigurableOptionsFragment>
+}
+
+export function ConfigurableProductUrls(props: ConfigurableProductUrlsProps) {
+  const { product, index = 0 } = props
+  const router = useRouter()
```

Comment:
This will reduce retriggers of the useEffect because the replace never changes.

```suggestion
  const { replace } = useRouter()
```